File: 2299-strong-password-checker-ii/2299-strong-password-checker-ii.cpp
class Solution {
public:
    bool strongPasswordCheckerII(string password) {
        if (password.length() < 8) { return false; }

        bool lowerCase = false;
        bool upperCase = false;
        bool digit = false;
        bool special = false;
        bool adj = false;

        for (int i = 0; i < password.length(); ++i) {
            char ch = password[i];
            if (ch >= '0' && ch <= '9') { digit = true; }
            else if (ch >= 'a' && ch <= 'z') { lowerCase = true; }
            else if (ch >= 'A' && ch <= 'Z') { upperCase = true; }
            else if (string("!@#$%^&*()-+").find(ch) != string::npos) { special = true; }
            
            if (i > 0 && ch == password[i - 1]) { adj = true; }
        }

        return lowerCase && upperCase && digit && special && !adj;
    }
};

File: 2299-strong-password-checker-ii/2299-strong-password-checker-ii.java
class Solution {
    public boolean strongPasswordCheckerII(String password) {
        if (password.length() < 8) { return false; }
        for (int i = 0; i < password.length() - 1; ++i) {
            if (password.charAt(i) == password.charAt(i + 1)) {
                return false;
            }
        }

        int countDigit = 0, countUpperCase = 0, countLowerCase = 0, countSpecialChar = 0;;
        for (int i = 0; i < password.length(); ++i) {
            char ch = password.charAt(i);
            if (Character.isLowerCase(ch)) {
                countLowerCase++;
            } else if (Character.isUpperCase(ch)) {
                countUpperCase++;
            } else if (Character.isDigit(ch)) {
                countDigit++;
            } else if (ch == '!' || ch == '@' || ch == '#' || ch == '$' || ch == '%' || ch == '^'
                        || ch == '&' || ch == '*' || ch == '(' || ch == ')' || ch == '-' || ch == '+') {
                countSpecialChar++;      
            }
        }

        if (countDigit == 0 || countUpperCase == 0 || countLowerCase == 0 || countSpecialChar == 0) { return false; }
        return true;
    }
}